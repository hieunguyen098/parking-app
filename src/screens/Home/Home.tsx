import { StyleSheet, ScrollView, SafeAreaView, KeyboardAvoidingView, View } from 'react-native';
import React from 'react';
import Header from './components/Header';
import ListVehicle from './components/ListVehicle';
import MenuBar from './components/MenuBar';

const Home = ({ navigation }: any) => {
    return (
        <SafeAreaView>
            <KeyboardAvoidingView behavior="position">
                <ScrollView style={styles.container}>
                    <Header />
                    <View style={styles.content}>
                        <MenuBar />
                        <ListVehicle />
                    </View>
                </ScrollView>
            </KeyboardAvoidingView>
        </SafeAreaView>
    );
};

export default Home;

const styles = StyleSheet.create({
    container: {},
    content: {
        marginHorizontal: 16,
    },
});
