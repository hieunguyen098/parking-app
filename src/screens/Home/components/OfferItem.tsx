import { Image, StyleSheet, Text, View } from 'react-native';
import React from 'react';

const OfferItem = () => {
    return (
        <View style={styles.container}>
            <Image
                style={styles.image}
                source={{ uri: 'http://img.giftpop.vn/brand/GIFTPOP/MP1807250002_BASIC_origin.jpg' }}
            />
            <Text style={styles.text}>Ưu đãi 30.000 hấp dẫn!! Nhanh tay nhận voucher</Text>
        </View>
    );
};

export default OfferItem;

const styles = StyleSheet.create({
    container: { marginBottom: 16 },
    image: {
        width: '100%',
        height: 200,
        marginVertical: 8,
        borderRadius: 8,
    },
    text: {
        fontSize: 20,
        fontWeight: '400',
        color: '#404040',
    },
});
