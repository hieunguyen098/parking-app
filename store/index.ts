import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './slices/authSlice';
import { generalReducer } from './slices/generalSlice';

export const store = configureStore({
    reducer: {
        auth: authReducer,
        general: generalReducer,
    },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
