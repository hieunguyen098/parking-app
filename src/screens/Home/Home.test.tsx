import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

import OfferItem from './components/OfferItem';
import { QueryClient, QueryClientProvider } from 'react-query';
import NewOffer from './components/NewOfferList';
import Home from './Home';
import { Provider } from 'react-redux';
import { store } from '../../../store';

jest.useFakeTimers();
test('test home', () => {
    const queryClient = new QueryClient();
    const component = (
        <QueryClientProvider client={queryClient}>
            <Provider store={store}>
                <NavigationContainer>
                    <Home />
                </NavigationContainer>
            </Provider>
        </QueryClientProvider>
    );

    const tree = render(component);
    expect(tree).toMatchSnapshot();
});
