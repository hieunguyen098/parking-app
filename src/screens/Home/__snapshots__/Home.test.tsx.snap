// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test home 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      stickyHeaderIndices={
        Array [
          1,
        ]
      }
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4294939962,
                16749882,
              ],
              "endPoint": undefined,
              "locations": undefined,
              "startPoint": undefined,
            }
          }
          style={
            Object {
              "paddingBottom": 32,
              "paddingHorizontal": 16,
              "paddingTop": 8,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 20,
                  "fontWeight": "600",
                }
              }
            >
              Chào, 
               !
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginRight": 8,
                    "position": "relative",
                  }
                }
              >
                <Text />
                <Text
                  style={
                    Object {
                      "backgroundColor": "red",
                      "borderRadius": 50,
                      "color": "white",
                      "fontSize": 12,
                      "paddingHorizontal": 5,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  3
                </Text>
              </View>
              <Image
                source={
                  Object {
                    "uri": "https://media.gq.com/photos/56bcb218cdf2db6945d2ef93/master/pass/bieber-coverstory-square.jpg",
                  }
                }
                style={
                  Object {
                    "borderRadius": 16,
                    "height": 32,
                    "width": 32,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "marginVertical": 10,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://cdn.vectorstock.com/i/1000x1000/75/78/parking-conceptual-web-banner-car-park-vector-12827578.webp",
                }
              }
              style={
                Object {
                  "borderRadius": 10,
                  "height": 100,
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderRadius": 10,
                "elevation": 2,
                "flexDirection": "row",
                "height": 44,
                "marginVertical": 10,
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <Text />
            <TextInput
              placeholder="Tìm kiếm bãi đỗ"
              placeholderTextColor="#FF953A"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 20,
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </ViewManagerAdapter_ExpoLinearGradient>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 32,
              "paddingHorizontal": 24,
              "paddingVertical": 8,
              "zIndex": 1,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 24,
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FF953A",
                    "fontSize": 12,
                    "fontWeight": "400",
                  },
                  undefined,
                ]
              }
            >
              Bãi đỗ xe
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 24,
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FF953A",
                    "fontSize": 12,
                    "fontWeight": "400",
                  },
                  undefined,
                ]
              }
            >
              Vé tháng
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 24,
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FF953A",
                    "fontSize": 12,
                    "fontWeight": "400",
                  },
                  undefined,
                ]
              }
            >
              Bạn bè
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 24,
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FF953A",
                    "fontSize": 12,
                    "fontWeight": "400",
                  },
                  undefined,
                ]
              }
            >
              Thanh toán
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "marginHorizontal": 16,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 16,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 24,
                      "fontWeight": "700",
                    },
                    undefined,
                  ]
                }
              >
                Danh sách xe gửi
              </Text>
              <View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#FF953A",
                        "fontSize": 16,
                        "fontWeight": "600",
                      }
                    }
                  >
                    Xem thêm
                  </Text>
                  <Text />
                </View>
              </View>
            </View>
            <ActivityIndicator
              size="large"
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 16,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 24,
                      "fontWeight": "700",
                    },
                    undefined,
                  ]
                }
              >
                Ưu đãi mới
              </Text>
              <View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#FF953A",
                        "fontSize": 16,
                        "fontWeight": "600",
                      }
                    }
                  >
                    Xem thêm
                  </Text>
                  <Text />
                </View>
              </View>
            </View>
            <Text>
              Loading
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
